src/pim-types.ts

```diff
@@ -255,6 +264,57 @@ export class TaskType extends EventType {
     ret.color = this.color;
     return ret;
   }
+
+  public getNextOccurence(): TaskType | null {
+    if (!this.isRecurring()) {
+      return null;
+    }
+
+    const rrule = this.rrule.clone();
+
+    if (rrule.count && rrule.count <= 1) {
+      return null; // end of reccurence
+    }
+
+    rrule.count = null; // clear count so we can iterate as many times as needed
+    const recur = rrule.iterator(this.startDate ?? this.dueDate);
+    let next = recur.next();
+    while ((next = recur.next())) {
+      if (next.compare(ICAL.Time.now()) > 0) {
+        break;
+      }
+    }
+
+    if (!next) {
+      return null; // end of reccurence
+    }
+
+    const nextStartDate = this.startDate && next;
+    const nextDueDate = this.dueDate && next;
+    if (nextStartDate && nextDueDate) {
+      const offset = (this.dueDate as ICAL.Time).subtractDateTz(this.startDate);
+      nextDueDate.addDuration(offset);
```

Comment:
Maybe I'm missing something, but isn't nextDueDate a boolean? How is this even going forward?

Reply:
No, it's either a `Time` (if start/dueDate exist) or it's undefined. I only need to calculate the offset if both are defined.

Reply:
Yeah, my bad.